modules/javafx.graphics/src/test/java/test/javafx/scene/layout/RegionTest.java

```diff
@@ -1282,6 +1282,17 @@ public void snapFunctionsShouldHandleExtremelyLargeValuesWithoutReturningNaN() {
         assertEquals(Long.MIN_VALUE, region.snapSpaceX(Double.NEGATIVE_INFINITY), 0.0);
         assertEquals(Long.MIN_VALUE, region.snapSpaceY(Double.NEGATIVE_INFINITY), 0.0);
 
+        // Inner space
+        assertEquals(Double.POSITIVE_INFINITY, region.snapInnerSpaceX(Double.MAX_VALUE), 0.0);
+        assertEquals(Double.POSITIVE_INFINITY, region.snapInnerSpaceY(Double.MAX_VALUE), 0.0);
+        assertEquals(Double.NEGATIVE_INFINITY, region.snapInnerSpaceX(-Double.MAX_VALUE), 0.0);
+        assertEquals(Double.NEGATIVE_INFINITY, region.snapInnerSpaceY(-Double.MAX_VALUE), 0.0);
```

Comment:
Why is the expected value infinity here?

Reply:
Good question!

ScaledMath.floor() goes through 
```
Math.floor(d + Math.ulp(d)) / scale;
```
which results in a value greater than Double.MAX_VALUE for positive input.

Similarly, the negative input goes through
```
Math.ceil(d - Math.ulp(d)) / scale;
```
and the result is less than -Double.MAX_VALUE, so negative infinity.

It's not really important because all these are way beyond the values acceptable as coordinates.